### PR TITLE
feat: merge progress data workers into signal worker

### PR DESCRIPTION
**Pull request by mstykow**
### Summary of changes

- merge progress data workers into signals worker to save on memory (they mostly need the same input data)
- delete any remnants of separate workers and their context providers

### Context and reason for change

#2384 

### How can the changes be tested

As you're switching between resources in audit view and check that the progress bars are still working as expected, look at the memory tab in the devtools to see that the memory footprint has been reduced compared to the base branch.

### CODE REVIEW

File: src/Frontend/web-workers/signals-worker.ts [hunk 1]
```diff
@@ -61,9 +77,11 @@ export class SignalsWorker {
 
   public processInput(input: SignalsWorkerInput) {
     this.setData(input);
-    this.dispatchAutocompleteSignals(input);
+    this.dispatchOverallProgressData(input);
+    this.dispatchFolderProgressData(input);
     this.dispatchAttributionsInFolderContent(input);
     this.dispatchSignalsInFolderContent(input);
+    this.dispatchAutocompleteSignals(input);
```
Comment (MarkusObendrauf) [inline src/Frontend/web-workers/signals-worker.ts:84]: we should re-arrange this to do the most important stuff first. suggestion:
```
    this.dispatchSignalsInFolderContent(input);
    this.dispatchAttributionsInFolderContent(input);
    this.dispatchFolderProgressData(input);
    this.dispatchOverallProgressData(input);
    this.dispatchAutocompleteSignals(input);
```

**Review by MarkusObendrauf — CHANGES_REQUESTED** [review]
Nice change! Performance is good.

File: src/Frontend/web-workers/signals-worker.ts (hunk 1 above)
Comment (mstykow) [reply src/Frontend/web-workers/signals-worker.ts:84]: i already ordered these items the way i thought they are the most important. my reason for putting the progress data first was that i can see the top progress bar first, independent of the selected resource. but thinking about it again now, i think that you're right that computing the signals for the middle column is more important so i'll change it.

File: src/Frontend/Components/ProgressBar/__tests__/TopProgressBar.test.tsx [hunk 2]
```diff
@@ -2,238 +2,84 @@
...
+          filesWithOnlyExternalAttributionCount: 3,
+          filesWithOnlyPreSelectedAttributionCount: 1,
+          resourcesWithMediumCriticalExternalAttributions: [],
+          resourcesWithNonInheritedExternalAttributionOnly: [],
+          resourcesWithHighlyCriticalExternalAttributions: [],
+        }),
+      ],
+    });
+
+    await userEvent.click(screen.getByRole('checkbox'), {
```
[the remark this answers is not in the document]
Comment (MarkusObendrauf) [reply src/Frontend/Components/ProgressBar/__tests__/TopProgressBar.test.tsx:63]: I still dont like it but ill live with it

**Review by MarkusObendrauf — APPROVED** [review]

**Pull request merged**